### DISCUSSION TITLE
Write large file to minio - best practice

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#18581] [Q&A] [answered]
Hi, what is the most appropriate way to write a large gzip-compressed CSV file, that I have downloaded from an API, to minio. Should I be looking at the s3 pickle io manager, or simply write code to push this file directly using the minio client package?

P.S. Im very new to dagster and it looks great.

**Comment by yuhan** [comment, marked as answer]
I'd recommend writing code to push this file directly using minio client package. While I/O manager is handy, it loads data into dataframe (in memory), so it may not fit your large scale usage.

**Reply by the4thamigo-uk** [reply]
Ah right, I see what you mean. Thanks.
